File: MPChartExample/src/com/xxmassdeveloper/mpchartexample/notimportant/DemoBase.java

package com.xxmassdeveloper.mpchartexample.notimportant;

import android.support.v4.app.FragmentActivity;

import com.xxmassdeveloper.mpchartexample.R;

/**
 * Baseclass of all Activities of the Demo Application.
 * 
 * @author Philipp Jahoda
 */
public abstract class DemoBase extends FragmentActivity {

    protected String[] mMonths = new String[] {
            "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Okt", "Nov", "Dec"
    };

    protected String[] mParties = new String[] {
            "Party A", "Party B", "Party C", "Party D", "Party E", "Party F", "Party G", "Party H",
            "Party I", "Party J", "Party K", "Party L", "Party M", "Party N", "Party O", "Party P",
            "Party Q", "Party R", "Party S", "Party T", "Party U", "Party V", "Party W", "Party X",
            "Party Y", "Party Z"
    };


//    protected ArrayList<XAxisValue> getMonths() {
//
//        ArrayList<XAxisValue> m = new ArrayList<XAxisValue>();
//        m.add(new XAxisValue(0, "Jan"));
//        m.add(new XAxisValue(1, "Feb"));
//        m.add(new XAxisValue(2, "Mar"));
//        m.add(new XAxisValue(3, "Apr"));
//        m.add(new XAxisValue(4, "May"));
//        m.add(new XAxisValue(5, "Jun"));
//        m.add(new XAxisValue(6, "Jul"));
//        m.add(new XAxisValue(7, "Aug"));
//        m.add(new XAxisValue(8, "Sep"));
//        m.add(new XAxisValue(9, "Okt"));
//        m.add(new XAxisValue(10, "Nov"));
//        m.add(new XAxisValue(11, "Dec"));
//
//        return m;
//    }
//
//
//    protected ArrayList<XAxisValue> getQuarters() {
//
//        ArrayList<XAxisValue> q = new ArrayList<XAxisValue>();
//        q.add(new XAxisValue(0, "Quarter 1"));
//        q.add(new XAxisValue(1, "Quarter 2"));
//        q.add(new XAxisValue(2, "Quarter 3"));
//        q.add(new XAxisValue(3, "Quarter 4"));
//
//        return q;
//    }
//
//    protected List<XAxisValue> getYears() {
//        ArrayList<XAxisValue> years = new ArrayList<XAxisValue>();
//
//        years.add(new XAxisValue(0, "2013"));
//        years.add(new XAxisValue(1, "2014"));
//        years.add(new XAxisValue(2, "2015"));
//        years.add(new XAxisValue(3, "2016"));
//        years.add(new XAxisValue(4, "2017"));
//        years.add(new XAxisValue(5, "2018"));
//        years.add(new XAxisValue(6, "2019"));
//
//        return years;
//    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        overridePendingTransition(R.anim.move_left_in_activity, R.anim.move_right_out_activity);
    }
}

File: MPChartExample/src/main/java/com/xxmassdeveloper/mpchartexample/notimportant/DemoBase.kt
package com.xxmassdeveloper.mpchartexample.notimportant

import android.Manifest
import android.content.pm.PackageManager
import android.graphics.Typeface
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import com.github.mikephil.charting.charts.Chart
import com.google.android.material.snackbar.Snackbar
import com.xxmassdeveloper.mpchartexample.R
import java.text.DateFormatSymbols

abstract class DemoBase : AppCompatActivity(), ActivityCompat.OnRequestPermissionsResultCallback {

    @JvmField
    protected val parties: Array<String> = arrayOf(
        "Party A", "Party B", "Party C", "Party D", "Party E", "Party F", "Party G", "Party H",
        "Party I", "Party J", "Party K", "Party L", "Party M", "Party N", "Party O", "Party P",
        "Party Q", "Party R", "Party S", "Party T", "Party U", "Party V", "Party W", "Party X",
        "Party Y", "Party Z"
    )

    @JvmField
    protected var tfRegular: Typeface? = null

    @JvmField
    protected var tfLight: Typeface? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        optionMenus.clear()

        tfRegular = Typeface.createFromAsset(assets, "OpenSans-Regular.ttf")
        tfLight = Typeface.createFromAsset(assets, "OpenSans-Light.ttf")
    }

    override fun onPrepareOptionsMenu(menu: Menu?): Boolean {
        menu?.let {
            for (i in 0 until menu.size()) {
                val menuItem: MenuItem = menu.getItem(i)
                optionMenus.add(menuItem.title.toString())
            }
        }
        return super.onPrepareOptionsMenu(menu)
    }

    override fun onBackPressed() {
        super.onBackPressed()
        overridePendingTransition(R.anim.move_left_in_activity, R.anim.move_right_out_activity)
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (requestCode == PERMISSION_STORAGE) {
            if (grantResults.size == 1 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                saveToGallery()
            } else {
                Toast.makeText(applicationContext, "Saving FAILED!", Toast.LENGTH_SHORT).show()
            }
        }
    }

    protected fun requestStoragePermission(view: View?) {
        if (ActivityCompat.shouldShowRequestPermissionRationale(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
            Snackbar.make(view!!, "Write permission is required to save image to gallery", Snackbar.LENGTH_INDEFINITE)
                .setAction(android.R.string.ok) {
                    ActivityCompat.requestPermissions(
                        this@DemoBase,
                        arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
                        PERMISSION_STORAGE
                    )
                }
                .show()
        } else {
            Toast.makeText(applicationContext, "Permission Required!", Toast.LENGTH_SHORT).show()
            ActivityCompat.requestPermissions(this@DemoBase, arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE), PERMISSION_STORAGE)
        }
    }

    protected fun saveToGallery(chart: Chart<*>?, name: String) {
        chart?.let {
            if (chart.saveToGallery(name + "_" + System.currentTimeMillis(), 70))
                Toast.makeText(applicationContext, "Saving SUCCESSFUL!", Toast.LENGTH_SHORT).show()
            else
                Toast.makeText(applicationContext, "Saving FAILED!", Toast.LENGTH_SHORT).show()
        }
    }

    protected abstract fun saveToGallery()

    companion object {
        private const val PERMISSION_STORAGE = 0
        //  Jan, Feb,... Dec
        val months = DateFormatSymbols().months.toList().map { it.take(3) }
        val optionMenus: MutableList<String> = mutableListOf()
    }
}
